flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointSubsumeHelper.java

```diff
@@ -87,6 +87,8 @@ private static boolean canSubsume(
             Optional<CompletedCheckpoint> latestNonSavepoint) {
         if (next == latest) {
             return false;
+        } else if (next.getProperties().isSavepoint()) {
+            return true;
```

Comment:
Add a unit test for this case?

Reply:
This is already covered by `CheckpointCoordinatorTest.testTriggerAndConfirmSimpleSavepoint`.